staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go

```diff
@@ -21,48 +21,46 @@ import (
 
 	k8smetrics "k8s.io/component-base/metrics"
 	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
 )
 
 var (
-	once sync.Once
-)
-
-// ControllerMetrics includes all the metrics of the proxy server.
-type ControllerMetrics struct {
 	controllerInstanceCount *k8smetrics.GaugeVec
-}
+	once                    sync.Once
+	manager                 string
+)
 
-// NewControllerMetrics create a new ControllerMetrics, configured with default metric names.
-func NewControllerMetrics() *ControllerMetrics {
-	controllerInstanceCount := k8smetrics.NewGaugeVec(
-		&k8smetrics.GaugeOpts{
-			Name:           "running_managed_controllers",
-			Help:           "Indicates where instances of a controller are currently running",
-			StabilityLevel: k8smetrics.ALPHA,
-		},
-		[]string{"name", "manager"},
-	)
-	controllerMetrics := &ControllerMetrics{
-		controllerInstanceCount: controllerInstanceCount,
-	}
+// InitializeControllerMetrics prepares metrics with specific manager.
+func InitializeControllerMetrics(manager string) {
 	once.Do(func() {
-		controllerMetrics.Register()
+		controllerInstanceCount = k8smetrics.NewGaugeVec(
```

Comment:
We generally:
- define metrics as vars
- do registration in Register() function

take a look at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go#L54 as an example

Reply:
Thanks for the pointers and discussion. I've changed code to follow suggested approach.